app/Http/Controllers/Api/UserController.php

```diff
@@ -251,6 +251,13 @@ public function store(Request $request)
             'email' => ['required', 'string', 'email', 'max:64', 'unique:users'],
             'password' => ['required', 'string', 'min:8', 'max:191'],
         ]);
+        
+        // Prevent the creation of new users via API if this is enabled.
+        if (!config('SETTINGS::SYSTEM:CREATION_OF_NEW_USERS', 'true')) {
+            throw ValidationException::withMessages([
+                'error' => "The creation of new users has been blocked by the system administrator."
+            ]);
+        } 
```

Comment:
Make it impossible to create users via the API if the option is disabled on the front-end.